assets/components/video/video.jsx

```diff
@@ -0,0 +1,68 @@
+// @flow
+
+// ----- Imports ----- //
+
+import React from 'react';
+
+import type { GridImg } from '../gridImage/gridImage';
+import GridImage from '../gridImage/gridImage';
+
+
+// ----- Constants ----- //
+
+const YOUTUBE_BASE = 'https://www.youtube.com/embed/';
+const YOUTUBE_PARAMS = 'enablejsapi=1&wmode=transparent&rel=0';
+
+
+// ----- Catalogue ----- //
+
+const videoCatalogue: {
+  scottTrustExplained: string,
+} = {
+  scottTrustExplained: 'jn4wAy1Xs5M',
+};
+
+
+// ----- Types ----- //
+
+// Utility type: https://flow.org/en/docs/types/utilities/#toc-keys
+export type videoName = $Keys<typeof videoCatalogue>;
+
+type PropTypes = {
+  name: videoName,
+  poster: ?GridImg,
+};
+
+
+// ----- Functions ----- //
+
+function youtubeUrl(name: videoName) {
+
+  const youtubeId = videoCatalogue[name];
+  const url = new URL(youtubeId, YOUTUBE_BASE);
+
+  return `${url.toString()}?${YOUTUBE_PARAMS}`;
```

Comment:
Not that I'm [aware of](https://developer.mozilla.org/en/docs/Web/API/URL#Constructor), I would be happy if it did. I believe you can pull query params out of a url using [`URL.searchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams), but not apply them.

I'm also not aware that flow provides a URL type, but I'm happy to be proved wrong as that would be pretty useful.